httpx/_content_streams.py

```diff
@@ -402,3 +388,18 @@ def encode(
         return IteratorStream(iterator=data)
 
     raise TypeError(f"Unexpected type for 'data', {type(data)!r}")
+
+
+def encode_response(content: ResponseContent = None) -> ContentStream:
+    if content is None:
+        return ByteStream(b"")
+    elif isinstance(content, bytes):
+        return ByteStream(body=content)
+    elif hasattr(content, "__aiter__"):
+        content = typing.cast(typing.AsyncIterator[bytes], content)
+        return AsyncIteratorStream(aiterator=content)
```

Comment:
Maybe you were already aware, but a thing I just learned from reviewing encode/starlette#1041: `collections.abc.AsyncIterator`/`collections.abc.Iterator` (which the `typing` equivalents derive from) have [a subclasshook](https://github.com/python/cpython/blob/v3.8.5/Lib/_collections_abc.py#L185-L189) which I think means you could just do:
```python
    elif isinstance(content, typing.AsyncIterator):
          return AsyncIteratorStream(aiterator=content)
```

Also _technically_ you may be casting Iterables to _Iterators_ here by just checking for the presence of `__aiter__`, though it seems unlikely to cause problems.

Reply:
@tomchristie